### PR TITLE
zh-cn: init the translation of `LockManager.request()`

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Related issues and pull requests

part of #17542

### CODE REVIEW

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+{{domxref("LockManager")}} 接口的 **`request()`** 方法请求一个 {{domxref('Lock')}} 对象，该方法发参数指定返回对象的名称和特征。请求的 `Lock` 对象被传递给回调，而函数本身返回一个 {{jsxref('Promise')}}，在锁被释放后解析（或拒绝）回调的结果，或者在请求被中止时拒绝。
+
+`options` 参数的 `mode` 属性可以接收 `"exclusive"` 或 `"shared"`。
+
+当锁一次只能由一个代码实例持有时，请求 `"exclusive"` 锁。这适用于标签页和 worker 中的代码。使用它来表示对资源的互斥访问。当持有给定名称的 `"exclusive"` 锁时，不能持有其他同名的锁。
+
+当代码的多个实例可以共享对资源的访问时，请求 `"shared"` 锁。当持有给定名称的 `"shared"` 锁时，可以授予同名的其他 `"shared"` 锁，但不能持有或授予具有该名称的 `"exclusive"` 锁。
+
+这种共享/独占锁模式在数据库事务架构中很常见，例如允许多个并发读取器（每个读取器请求一个 `"shared"` 锁），但仅允许一个写入器（单个 `"exclusive"` 锁）。这称为读者——作者模式。在 [IndexedDB API](/zh-CN/docs/Web/API/IndexedDB_API) 中，这被公开为具有相同语义的 `"readonly"` 和 `"readwrite"` 事务。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:18]: 公开 ==> 暴露

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+    - `ifAvailable` {{optional_inline}}
+
+      - : 如果为 `true`，则只有在尚未持有锁定请求时才会授予该锁定请求。如果无法授予，则将传入 `null` 而非 `Lock` 实例来调用回调。默认值为 `false`。
+
+    - `steal` {{optional_inline}}
+
+      - : 如果为 `true`，则任何持有的同名锁将被释放，并且请求将被授予，抢占任何排队中的锁请求。默认值为 `false`。
+
+      > **警告：** 小心使用！之前在锁内运行的代码继续运行，并且可能与现在持有锁的代码发生冲突。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:51]:
```suggestion
        > **警告：** 小心使用！之前在锁内运行的代码会继续运行，并且可能与现在持有锁的代码发生冲突。
```

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+### 参数
+
+- `name`
+
+  - : 你想要请求的锁的标识符。
+
+- `options` {{optional_inline}}
+
+  - : 一个描述你要创建的锁的特征的对象。合法的值为：
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:37]:
```suggestion
  - : 一个描述你要创建的锁的特征的对象。有效的值为：
```

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      - : 如果为 `true`，则只有在尚未持有锁定请求时才会授予该锁定请求。如果无法授予，则将传入 `null` 而非 `Lock` 实例来调用回调。默认值为 `false`。
+
+    - `steal` {{optional_inline}}
+
+      - : 如果为 `true`，则任何持有的同名锁将被释放，并且请求将被授予，抢占任何排队中的锁请求。默认值为 `false`。
+
+      > **警告：** 小心使用！之前在锁内运行的代码继续运行，并且可能与现在持有锁的代码发生冲突。
+
+    - `signal` {{optional_inline}}
+      - : 一个 {{domxref("AbortSignal")}}（{{domxref("AbortController")}} 的 {{domxref("AbortController.signal", "signal")}} 属性）；如果指定并且 {{domxref("AbortController")}} 被中止，则锁请求将被抛弃（如果尚未授予）。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:54]:
```suggestion
      - : 一个 {{domxref("AbortSignal")}}（{{domxref("AbortController")}} 的 {{domxref("AbortController.signal", "signal")}} 属性）；如果指定并且 {{domxref("AbortController")}} 被中止，则锁请求将被丢弃（如果尚未授予）。
```

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+- `callback`
+  - ：授予锁定时调用的方法。当回调返回（或抛出异常）时，锁会自动释放。通常回调是一个异步函数，这会导致只有当异步函数完全完成时才会释放锁。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:57]:
```suggestion
  - ：授予锁时调用的方法。当回调返回（或抛出异常）时，锁会自动释放。通常回调是一个异步函数，这会导致只有当异步函数完全完成时才会释放锁。
```

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+### 异常
+
+此方法可能会返回一个被以下类型之一的 {{domxref("DOMException")}} 拒绝的 promise：
+
+- `InvalidStateError` {{domxref("DOMException")}}
+  - ：如果当前环境的文档未完全激活，则抛出该异常。
+- `SecurityError` {{domxref("DOMException")}}
+  - ：如果无法获取当前环境的锁管理器，则抛出该异常。
+- `NotSupportedError` {{domxref("DOMException")}}
+  - : 如果 `name` 以连字符 (`-`) 开头，或选项 `steal` 和 `ifAvailable` 均为 `true`，或者选项 `signal` 存在且选项 `steal` 或 `ifAvailable` 为 `true`。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:72]:
```suggestion
  - : 如果 `name` 以连字符（`-`）开头，或选项 `steal` 和 `ifAvailable` 均为 `true`，或者选项 `signal` 存在且选项 `steal` 或 `ifAvailable` 为 `true`。
```

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+await navigator.locks.request("my_resource", async (lock) => {
+  // 锁被授予。
+});
+```
+
+### `mode` 示例
+
+以下示例展示了如何为读取器和写入器使用 `mode` 选项。
+
+请注意，这两个函数都使用名为 `my_resource` 的锁。`do_read()` 请求 `'shared'` 模式下的锁定，这意味着多个调用可能会在不同的事件处理程序、标签页或 worker 中同时发生。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:92]:
```suggestion
请注意，这两个函数都使用名为 `my_resource` 的锁。`do_read()` 请求 `'shared'` 模式下的锁定，这意味着多个调用可能会在不同的事件处理器、标签页或 worker 中同时发生。
```

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+    "my_resource",
+    { mode: "shared" },
+    async (lock) => {
+      // 在这里放置执行读取器具体操作的代码。
+    },
+  );
+}
+```
+
+`do_write()` 函数使用相同的锁，但处于 `'exclusive'` 模式，这将延迟 `do_read()` 中 `request()` 的调用，直到写操作完成。这适用于事件处理程序、标签页或 worker。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:106]:
```suggestion
`do_write()` 函数使用相同的锁，但处于 `'exclusive'` 模式，这将延迟 `do_read()` 中 `request()` 的调用，直到写操作完成。这适用于事件处理器、标签页或 worker。
```

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    // 该锁已被授予，并且该源中没有其他正在运行的代码持有 'my_res_lock' 锁，直到该锁返回为止。
+  },
+);
+```
+
+### `signal` 示例
+
+如果只等待一小段时间的锁定，使用 `signal` 选项。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:141]:
```suggestion
如果只等待锁一小段时间，请使用 `signal` 选项。
```

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+  await navigator.locks.request(
+    "my_resource",
+    { signal: controller.signal },
+    async (lock) => {
+      // 锁被获取了！
+    },
+  );
+} catch (ex) {
+  if (ex.name === "AbortError") {
+    // 该请求在获得批准之前就中止了。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:158]: How about this?

```suggestion
    // 该请求在获得锁之前就中止了。
```

File: files/zh-cn/web/api/lockmanager/request/index.md (hunk 1 above)
Comment (skyclouds2001) [inline files/zh-cn/web/api/lockmanager/request/index.md:18]:
```suggestion
这种共享/独占锁模式在数据库事务架构中很常见，例如允许多个并发读取器（每个读取器请求一个 `"shared"` 锁），但仅允许一个写入器（单个 `"exclusive"` 锁）。这称为读者——作者模式。在 [IndexedDB API](/zh-CN/docs/Web/API/IndexedDB_API) 中，这被暴露为具有相同语义的 `"readonly"` 和 `"readwrite"` 事务。
```

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+
+      - : 如果为 `true`，则任何持有的同名锁将被释放，并且请求将被授予，抢占任何排队中的锁请求。默认值为 `false`。
+
+        > **警告：** 小心使用！之前在锁内运行的代码会继续运行，并且可能与现在持有锁的代码发生冲突。
+
+    - `signal` {{optional_inline}}
+      - : 一个 {{domxref("AbortSignal")}}（{{domxref("AbortController")}} 的 {{domxref("AbortController.signal", "signal")}} 属性）；如果指定并且 {{domxref("AbortController")}} 被中止，则锁请求将被丢弃（如果尚未授予）。
+
+- `callback`
+  - ：授予锁时调用的方法。当回调返回（或抛出异常）时，锁会自动释放。通常回调是一个异步函数，这会导致只有当异步函数完全完成时才会释放锁。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:57]:
```suggestion
  - : 授予锁时调用的方法。当回调返回（或抛出异常）时，锁会自动释放。通常回调是一个异步函数，这会导致只有当异步函数完全完成时才会释放锁。
```

File: files/zh-cn/web/api/lockmanager/request/index.md [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+此方法可能会返回一个被以下类型之一的 {{domxref("DOMException")}} 拒绝的 promise：
+
+- `InvalidStateError` {{domxref("DOMException")}}
+  - ：如果当前环境的文档未完全激活，则抛出该异常。
+- `SecurityError` {{domxref("DOMException")}}
+  - ：如果无法获取当前环境的锁管理器，则抛出该异常。
+- `NotSupportedError` {{domxref("DOMException")}}
+  - : 如果 `name` 以连字符（`-`）开头，或选项 `steal` 和 `ifAvailable` 均为 `true`，或者选项 `signal` 存在且选项 `steal` 或 `ifAvailable` 为 `true`。
+- `AbortError` {{domxref("DOMException")}}
+  - ：如果选项 `signal` 存在并且被中止，则抛出该异常。
```
Comment (yin1999) [inline files/zh-cn/web/api/lockmanager/request/index.md:74]:
```suggestion
  - : 如果选项 `signal` 存在并且被中止，则抛出该异常。
```

**Review by yin1999 — APPROVED** [review]
Thank you @skyclouds2001

**Pull request merged**